tasks/broad/ImputationBeagleTasks.wdl

```diff
@@ -1,5 +1,47 @@
 version 1.0
 
+task InferContigsFromVcf {
+  input {
+    File vcf_input
+    Array[String] allowed_contigs
+    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.5.0.0"
+    Int cpu = 1
+    Int memory_mb = 16000
+    Int disk_size_gb = ceil(size(vcf_input, "GiB")) + 10
+  }
+
+  command <<<
+    set -e -o pipefail
+
+    # extract all chromosomes in input
+    gunzip -c ~{vcf_input} | grep -v "#" | cut -f1 | sort -u > chromosomes.txt
+    echo "Extracted chromosomes: $(cat chromosomes.txt | tr '\n' ' ')"
+
+    # filter to allowed contigs, enforcing order
+    for chr in ~{sep=" " allowed_contigs}; do
+        if grep -q "^${chr}$" "chromosomes.txt"; then
+            echo "${chr}" >> filtered_chromosomes.txt
+        fi
+    done
+    
+    echo "Filtered chromosomes: $(cat filtered_chromosomes.txt | tr '\n' ' ')"
```

Comment:
I kind of feel like this should error out if things in the allow list got filtered or is that too strict?

Reply:
ignore after faces chat! (i had a bad understanding)